File: Euclid/Messages/Incoming/PURCHASE.cs
using Euclid.Game;
using Euclid.Messages.Outgoing;
using Euclid.Network.Streams;
using Euclid.Storage.Database.Access;
using Euclid.Storage.Database.Data;
using Euclid.Util.Extensions;
using System;

namespace Euclid.Messages.Incoming
{
    class PURCHASE : IMessageEvent
    {
        public void Handle(Player player, Request request)
        {
            if (!CatalogueManager.Instance.OrderHistory.ContainsKey(player.Details.Id))
            {
                return;
            }

            var order = CatalogueManager.Instance.OrderHistory[player.Details.Id];
            CatalogueManager.Instance.OrderHistory.Remove(player.Details.Id);

            if (order == null)
            {
                return;
            }

            var itemData = new ItemData()
            {
                OwnerId = player.Details.Id,
                DefinitionId = order.Item.Definition.Data.Id,
                CustomData = order.CustomData
            };

            ItemDao.CreateItem(itemData);

            player.PrivateServer.Inventory.AddItem(new Item(itemData));
            player.Send(new Outgoing.ADDSTRIPITEM());
        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/PURCHASE.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.game.catalogue.CatalogueManager;
import euclid.game.catalogue.CatalogueOrder;
import euclid.game.item.Item;
import euclid.messages.IMessageEvent;
import euclid.messages.outgoing.ADDSTRIPITEM;
import euclid.network.streams.Request;
import euclid.storage.database.access.ItemDao;
import euclid.storage.database.data.ItemData;

public class PURCHASE implements IMessageEvent {

    @Override
    public void handle(Player player, Request request) {
        if (!CatalogueManager.getInstance().getOrderHistory().containsKey(player.getDetails().getId())) {
            return;
        }

        CatalogueOrder order = CatalogueManager.getInstance().getOrderHistory().get(player.getDetails().getId());
        CatalogueManager.getInstance().getOrderHistory().remove(player.getDetails().getId());

        if (order == null) {
            return;
        }

        ItemData itemData = new ItemData();
        itemData.setOwnerId(player.getDetails().getId());
        itemData.setDefinitionId(order.getItem().getDefinition().getData().getId());
        itemData.setCustomData(order.getCustomData());

        ItemDao.createItem(itemData);

        player.getInventory().addItem(new Item(itemData));
        player.send(new ADDSTRIPITEM());
    }
}
